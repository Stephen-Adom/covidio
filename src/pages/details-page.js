import React from 'react';

const Details = () => (
  <div>details-page</div>
);

export default Details;
