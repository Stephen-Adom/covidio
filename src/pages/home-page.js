import React from 'react';

const Home = () => (
  <div id="home"> Home</div>
);

export default Home;
