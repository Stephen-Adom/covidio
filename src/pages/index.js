export { default as Continent } from './details-page';
export { default as Home } from './home-page';
export { default as SplashScreen } from './splash-page';
