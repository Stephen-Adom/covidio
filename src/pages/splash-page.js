const SplashScreen = () => <div>SplashScreen</div>;
export default SplashScreen;
