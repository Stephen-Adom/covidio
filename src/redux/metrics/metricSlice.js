import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import {
  fetchAllCountries,
  fetchContinent,
  fetchGlobalData,
  fetchAllContinents,
} from '../api/endpoints';

const initialState = {
  global: {},
  countries: [],
  continent: {},
  allContinents: [],
  status: '',
  error: '',
};

export const globalInfo = createAsyncThunk('metrics/all', async (thunkAPI) => {
  try {
    const response = await fetchGlobalData();
    return response.data;
  } catch (err) {
    return thunkAPI.rejectWithValue(err.response.data);
  }
});

export const getAllContinents = createAsyncThunk('metrics/all-continents', async (thunkAPI) => {
  try {
    const response = await fetchAllContinents();
    return response.data;
  } catch (err) {
    return thunkAPI.rejectWithValue(err.response.data);
  }
});

export const getAllCountries = createAsyncThunk(
  'metrics/countries',
  async (thunkAPI) => {
    try {
      const response = await fetchAllCountries();
      return response.data;
    } catch (err) {
      return thunkAPI.rejectWithValue(err.response.data);
    }
  },
);

export const getContinent = createAsyncThunk(
  'metrics/continent',
  async (payload, thunkAPI) => {
    try {
      const response = await fetchContinent(payload);
      return response.data;
    } catch (err) {
      return thunkAPI.rejectWithValue(err.response.data);
    }
  },
);

const metricSlice = createSlice({
  name: 'metrics',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(globalInfo.pending, (state) => {
        const IsPending = state;
        IsPending.status = 'pending';
      })
      .addCase(globalInfo.fulfilled, (state, action) => {
        const IsFulfilled = state;
        IsFulfilled.status = 'fulfilled';
        IsFulfilled.global = action.payload;
        IsFulfilled.error = '';
      })
      .addCase(globalInfo.rejected, (state, action) => {
        const IsRejected = state;
        IsRejected.status = 'rejected';
        IsRejected.global = {};
        IsRejected.error = action.payload;
      })

      .addCase(getAllContinents.pending, (state) => {
        const IsPending = state;
        IsPending.status = 'pending';
      })

      .addCase(getAllContinents.fulfilled, (state, action) => {
        const IsFulfilled = state;
        IsFulfilled.status = 'fulfilled';
        IsFulfilled.allContinents = action.payload;
        IsFulfilled.error = '';
      })

      .addCase(getAllContinents.rejected, (state, action) => {
        const IsRejected = state;
        IsRejected.status = 'rejected';
        IsRejected.error = action.payload;
      })

      .addCase(getAllCountries.pending, (state) => {
        const IsPending = state;
        IsPending.status = 'pending';
      })
      .addCase(getAllCountries.fulfilled, (state, action) => {
        const IsFulfilled = state;
        IsFulfilled.status = 'fulfilled';
        IsFulfilled.countries = action.payload;
        IsFulfilled.error = '';
      })
      .addCase(getAllCountries.rejected, (state, action) => {
        const IsRejected = state;
        IsRejected.status = 'rejected';
        IsRejected.countries = [];
        IsRejected.error = action.payload;
      })
      .addCase(getContinent.pending, (state) => {
        const IsPending = state;
        IsPending.status = 'pending';
      })
      .addCase(getContinent.fulfilled, (state, action) => {
        const IsFulfilled = state;
        IsFulfilled.status = 'fulfilled';
        IsFulfilled.continent = action.payload;
        IsFulfilled.error = '';
      })
      .addCase(getContinent.rejected, (state, action) => {
        const IsRejected = state;
        IsRejected.status = 'rejected';
        IsRejected.continent = [];
        IsRejected.error = action.payload;
      });
  },
});

export default metricSlice.reducer;
