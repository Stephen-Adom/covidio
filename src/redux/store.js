import { configureStore } from '@reduxjs/toolkit';
import metricSlice from './metrics/metricSlice';

const store = configureStore({
  reducer: {
    metrics: metricSlice,
  },
});

export default store;
