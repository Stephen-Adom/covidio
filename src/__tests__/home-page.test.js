import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { RouterProvider } from 'react-router-dom';
import store from '../redux/store';
import router from '../route/routes';
import { Home } from '../pages';
import { globalInfo, getAllContinents } from '../redux/metrics/metricSlice';

describe('Home Page', () => {
  beforeEach(async () => {
    await store.dispatch(globalInfo());
    await store.dispatch(getAllContinents());
  });

  it('should render home page correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RouterProvider router={router}>
          <Home />
        </RouterProvider>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
