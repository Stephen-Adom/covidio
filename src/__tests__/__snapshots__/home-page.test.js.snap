// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render home page correctly 1`] = `
<div
  className="App"
>
  <section
    className="splash-container"
  >
    <div
      className="relative logo-container"
    >
      <img
        alt="covido"
        className="bounce-in-fwd"
        src="covido-logo.png"
        width="100%"
      />
      <h1
        className="logo tracking-in-expand"
      >
        Covidio
      </h1>
    </div>
  </section>
</div>
`;
